projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -151,6 +151,21 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
     @Output()
     public dimensionsChange = new EventEmitter<IDimensionsChange>();
 
+    /* blazorInclude */
+    /**
+     * Emitted when the the pivot configuration is changed.
+     *
+     * @internal
+     * @remarks
+     * Returns the new dimension collection and its type:
+     * @example
+     * ```html
+     * <igx-pivot-grid #grid [data]="localData" [height]="'305px'"
+     *              (dimensionsChange)="dimensionsChange($event)"></igx-grid>
+     * ```
+     */
+    @Output()
+    public configurationChange = new EventEmitter<IPivotConfigurationChange>();
```

Comment:
@gmurray81 would this also work as `<prop>Change`-type event that directly emits the config itself so it could also work with two way binding? Though it being an object makes it a bit iffy with immutability and change detection.

Reply:
Actually, there are already events for dimension and value changes, right? Assuming the other bit that can change through the UI would be rows and values? If we've started down that path, would it actually be useful to expose those too instead?

Reply:
yeah, if you wanted it to be public, you could make it a bananabox style event and that would work just fine for our purposes. We'd just need to change the attribute path that describes where to find the updated value.

yeah, I saw the other events, but for our purposes it would be useful to have an event that fires whenever anything changes in the pivot config, and that the args is the entire pivot config payload for us to marshal accross (unless we start trying to play games with synchronizing only portions of the tree when the grid updates them). I wanted to start with something reasonably brute force though, since I want to get to the point where this behaves acceptably for people that want to make updates both programatically and via the data selector. 

There's some remaining work to be done still possibly since we need to make sure that if we replicate new items coming from the data selector or the grid outward, that these can roundtrip back in appropriately.

This isn't necessarily straightforward though since the grid or selector may create new dimensions, I'm guessing, that have callback methods on them, without necessarily also setting an identifier that would allow for the callback to be fetched again if the dimension were round-tripped through blazor. The best option would be if we could make it always set enough information on the dimension that it can definitely round trip through blazor, failing that we could possibly consider something sneaky, whereby when a dimension is round tripping outward, and it has callback functions we need to ensure round trip, we could technically cache them somewhere and put an identifier in something like aggregatorScript, so we can fetch the callback function later when it round trips back.

I can't tell if I'm making sense or not :)